event-sourcing/event-source-api/src/main/java/uk/gov/justice/services/eventsourcing/source/api/service/EventStreamPageService.java

```diff
@@ -0,0 +1,152 @@
+        return pageOfEventStreamWithLinks(positionOf, pageSize, uriInfo, convertStreamIdToSelfLinks(eventStreamEntries, pageSize, uriInfo));
+    }
+
+    private List<EventStreamEntry> convertStreamIdToSelfLinks(final List<EventStreamEntry> eventStreamEntries,
+                                                              final long pageSize,
+                                                              final UriInfo uriInfo) throws MalformedURLException {
+        final List<EventStreamEntry> entriesWithSelfLinks = new ArrayList<>();
+
+        for (EventStreamEntry eventStreamEntry : eventStreamEntries) {
+            final Optional<URL> selfLink = selfLink(eventStreamEntry.getStreamId(), pageSize, uriInfo);
+            final EventStreamEntry eventStreamEntryWithSelfLink = new EventStreamEntry(selfLink.get().toString(), eventStreamEntry.getSequenceNumber());
+            entriesWithSelfLinks.add(eventStreamEntryWithSelfLink);
+        }
+        return entriesWithSelfLinks;
+    }
+
+    private Page<EventStreamEntry> pageOfEventStreamWithLinks(final Position position,
+                                                              final long pageSize,
+                                                              final UriInfo uriInfo,
+                                                              final List<EventStreamEntry> entries) throws MalformedURLException {
+        if (entries.isEmpty()) {
+            return emptyPageWithLinks(pageSize, uriInfo);
+        }
+
+        final long maxSequenceId = max(entries);
+        final long minSequenceId = min(entries);
+
+        final boolean newEventsAvailable = !position.isHead() && eventStreamService.recordExists(maxSequenceId + 1);
+        final boolean olderEventsAvailable = !position.isFirst() && eventStreamService.recordExists(minSequenceId - 1);
+
+        final PagingLinks.PagingLinksBuilder pagingLinksBuilder = new PagingLinks.PagingLinksBuilder(
+                fixedPositionPageHref(head(), BACKWARD, pageSize, uriInfo),
+                fixedPositionPageHref(first(), FORWARD, pageSize, uriInfo));
+
+        final Optional<URL> previous = newEventsAvailable ? pageHref(positionOf(maxSequenceId + 1), FORWARD, pageSize, uriInfo) : Optional.empty();
+
+        final Optional<URL> next = olderEventsAvailable ? pageHref(positionOf(minSequenceId - 1), BACKWARD, pageSize, uriInfo) : Optional.empty();
+
+        pagingLinksBuilder.previous(previous);
+
+        pagingLinksBuilder.next(next);
+
+        return new Page(entries, pagingLinksBuilder.build());
+    }
+
+    private Page<EventStreamEntry> emptyPageWithLinks(long pageSize, UriInfo uriInfo) throws MalformedURLException {
+        final PagingLinks.PagingLinksBuilder pagingLinksBuilder = new PagingLinks.PagingLinksBuilder(fixedPositionPageHref(head(), BACKWARD, pageSize, uriInfo),
+                fixedPositionPageHref(first(), FORWARD, pageSize, uriInfo));
+        return new Page<>(new ArrayList<>(), pagingLinksBuilder.build());
```

Comment:
return collections.emptylist not new array list?

Reply:
done